tools/python/util/build_helpers.py

```diff
@@ -0,0 +1,203 @@
+#  // Copyright (c) Microsoft Corporation. All rights reserved.
+#  // Licensed under the MIT License.
+import contextlib
+import sys
+import shutil
+import os
+from .UsageError import UsageError
+from .BuildError import BuildError
+from .platform_helpers import is_windows, is_macOS
+
+
+def version_to_tuple(version: str) -> tuple:
+    v = []
+    for s in version.split("."):
+        with contextlib.suppress(ValueError):
+            v.append(int(s))
+    return tuple(v)
+
+
+def check_python_version():
+    required_minor_version = 7
+    if (sys.version_info.major, sys.version_info.minor) < (3, required_minor_version):
+        raise UsageError(
+            f"Invalid Python version. At least Python 3.{required_minor_version} is required. "
+            f"Actual Python version: {sys.version}"
+        )
+
+
+def str_to_bool(s):
+    """Convert string to bool (in argparse context)."""
+    if s.lower() not in ["true", "false"]:
+        raise ValueError("Need bool; got %r" % s)
+    return {"true": True, "false": False}[s.lower()]
+
+
+def is_reduced_ops_build(args):
+    return args.include_ops_by_config is not None
+
+
+def resolve_executable_path(command_or_path):
+    """Returns the absolute path of an executable."""
+    if command_or_path and command_or_path.strip():
+        executable_path = shutil.which(command_or_path)
+        if executable_path is None:
+            raise BuildError(f"Failed to resolve executable path for '{command_or_path}'.")
+        return os.path.abspath(executable_path)
+    else:
+        return None
+
+
+def get_config_build_dir(build_dir, config):
+    # build directory per configuration
+    return os.path.join(build_dir, config)
+
+
+def use_dev_mode(args):
+    if args.compile_no_warning_as_error:
+        return False
+    if args.use_acl:
+        return False
+    if args.use_armnn:
+        return False
+    if args.ios and is_macOS():
+        return False
+    SYSTEM_COLLECTIONURI = os.getenv("SYSTEM_COLLECTIONURI")  # noqa: N806
+    if SYSTEM_COLLECTIONURI and SYSTEM_COLLECTIONURI != "https://dev.azure.com/onnxruntime/":
+        return False
+    return True
+
+
+def add_default_definition(definition_list, key, default_value):
+    for x in definition_list:
+        if x.startswith(key + "="):
+            return definition_list
+    definition_list.append(key + "=" + default_value)
+
+
+def normalize_arg_list(nested_list):
+    return [i for j in nested_list for i in j] if nested_list else []
+
+
+def number_of_parallel_jobs(args):
+    return os.cpu_count() if args.parallel == 0 else args.parallel
+
+
+def number_of_nvcc_threads(args):
+    if args.nvcc_threads >= 0:
+        return args.nvcc_threads
+
+    nvcc_threads = 1
+    import psutil
+    try:
+        available_memory = psutil.virtual_memory().available
```

Comment:
```suggestion
    
    try:
        import psutil
    except ImportError:
        print(
            "Failed to import psutil. Please `pip install psutil` for better estimation of nvcc threads. Use "
            "nvcc_threads=1"
        )
   
   available_memory = psutil.virtual_memory().available
```

I would keep the try-catch block small. Previously the import was not in the block so the catch was not meaningful.